compiler/optimizer.go

```diff
@@ -22,6 +22,12 @@ func (c *Compiler) Optimize(optLevel, sizeLevel int, inlinerThreshold uint) erro
 		c.replacePanicsWithTrap() // -panic=trap
 	}
 
+	// run a check of all of our code
+	c.check()
+	if err := c.Verify(); err != nil {
```

Comment:
The `Verify` here is not necessary, as the module has already been verified right before calling `Optimize`.
The pattern in most parts of the compiler is to verify right after doing a large transform on the module, not before. That also makes error messages easier to understand.